### PR TITLE
solutions for lecture 4

### PR DESCRIPTION
**Pull request by giacomocavalieri**
### Solutions for Lecture 4

cc @chshersh

### CODE REVIEW

File: src/Lecture4.hs [hunk 1]
```diff
@@ -135,7 +138,21 @@ errors. We will simply return an optional result here.
 -}
 
 parseRow :: String -> Maybe Row
-parseRow = error "TODO"
+parseRow string = do
+   [rawProd, rawTradeType, rawCost] <- pure $ splitOn ',' string
+   prod <- validate (not . null) rawProd -- product must be a non empty String
+   tradeType <- readMaybe rawTradeType -- type is one of the options
+   cost <- readMaybe rawCost >>= validate (> 0) -- cost must be > 0
+   pure $ Row prod tradeType cost
```
Comment (chshersh) [inline src/Lecture4.hs]: Perfect monadic validation 👨🏻‍🍳 🤌

**Review by chshersh — APPROVED** [review]
:tada: Fantastic work! Congratulations on finishing all **four** lectures! 🥳 
You're truly a Functional Adept now 🧠 
I hope you had fun :man_dancing: :microphone:

If you would like to receive a pretty certificate about the course completion, please, send your full name and the link to this PR to `haskell.beginners2022@gmail.com` 📧 

If you have time, I would appreciate your feedback on the course:

* [Haskell Beginners 2022 Course Feedback](https://docs.google.com/forms/d/e/1FAIpQLSfCkST64dnDCYkz5mezkVyLfWjeskSxhV0Y_cYxUaJw0U9ZhQ/viewform)

If you like my work, you can [sponsor me on GitHub](https://github.com/sponsors/chshersh) 💖

You can also ask any questions here or anywhere else and we can discuss any topics you would like to dig into as well  :slightly_smiling_face: Don't be a stranger!

**Comment by giacomocavalieri** [comment]
I would just like to thank you so much for this amazing project! It was great fun and your comments were always nice and useful :)

**Comment by chshersh** [comment]
I'm glad you enjoyed the course and my feedback! I'm always happy to hear such warm words about my work 🤗 😊

**Pull request merged**